e2eAIOK/deltatuner/README.md

```diff
@@ -0,0 +1,63 @@
+Deltatuner
+============
+Deltatuner is an extension for [Peft](https://github.com/huggingface/peft) to improve LLM fine-tuning speed through multiple optimizations, including: leverage the compact model constructor [DE-NAS](https://github.com/intel/e2eAIOK/tree/main/e2eAIOK/DeNas) to construct/modify the compact delta layers in a hardware-aware and train-free approach, and adding more new deltatuning algorithms.
+
+![Architecure](./doc/deltatuner.png)
+
+Below is an example showing how to optimize the finetuning process of [Intel NeuralChat](https://github.com/intel/intel-extension-for-transformers/tree/main/workflows/chatbot/fine_tuning).
+
+# Installation
+- install the python package
+```shell
+git clone https://github.com/intel-innersource/frameworks.bigdata.AIDK.git deltatuner
+cd deltatuner
+git checkout deltatuner
+pip install -e .
+```
+
+
+# Fine-tuning Use Cases
+
+We use delatuner to optimize the [LoRA approach](https://arxiv.org/pdf/2106.09685.pdf) to finetune the LLM efficiently. 
+
+## Fine-tuning on MPT-7B
+For [MPT](https://huggingface.co/mosaicml/mpt-7b), adding the following command to use the delatuner optimizations in the [LLM fine-tuning script](./example/instruction_tuning_pipeline/finetune_clm.py) for finetuning on the Alpaca dataset. 
+
```

Comment:
This code block example not working well, where should the user add it? if the intention is to showcase limited code of changes, please refine.

Reply:
This example is a sample that aims to demonstrate the easy usage of deltatuner by adding a few lines, and it can work without adding extra codes to only use the deltatuner optimization. If the user want to train the optimized model, and he can add some training codes after the above example as shown in the example folder. The scripts in the example folder show the completed process of detatuner-optimization, training and evaluation.